Integrate internalisation in setup script

jesse5555 opened the pull request:
gettext gives us the ability to embrace python's internalisation by
creating POT files and setting translatable strings in out setup-script.

jesse5555 commented on init-odkx-sync-endpoint.py:
```diff
@@ -11,31 +11,36 @@
     try:
         env = parse_env_file(env_file_location)
-        typer.echo(f"Found configuration at {env_file_location}")
+        typer.echo(_("Found configuration at {env_file_location}").format(env_file_location=env_file_location))
```
hey @matt9j @wbrunette, I landed on some resource https://www.mattlayman.com/blog/2015/i18n/ that opened my eyes alittle about how all this might work.

However, i am not sure if `_("some string {value}").formart(value=some-value)` is a right way to setup a transferable string.

matt9j replied:
I think this is a reasonable approach for now! The string will appear in the .po file as "some string {value}" and the translation will just need to keep the value format parameter intact.

jesse5555 commented on locale/en/LC_MESSAGES/messages.po:
```diff
@@ -0,0 +1,17 @@
+# ! /usr/bin/env python3
+# Compute absolute paths from relative path inputs
+msgid ""
```
i am using babel to generate these files but there is no way i can generate these translatable strings with the command `pybabel compile -i init-odkx-sync-endpoint.py -l en -d locale`

matt9j replied:
I think that you need to be using the "extract" command to generate the .po file first, then once you have a .po file, you "compile" it to a .mo file! Try the command `pybabel extract -o locale/en/LC_MESSAGES/messages.po ./` instead, and I think it should work on your machine to fill the .po file.

jesse5555 replied:
this actually worked @matt9j 
Thank you so much

jesse5555 reviewed (commented):
Hey  @matt9j @wbrunette!

This has not yet worked.  compiling the script to generate  `.po` files is producing errors for now 
```
WARNING: Problem on line 197: '"""'
WARNING: Start of line didn't match any expected keyword.
WARNING: Problem on line 198: 'with open(filepath, mode="w") as f:'
WARNING: Start of line didn't match any expected keyword.
WARNING: Problem on line 199: 'for (key, val) in env.items():'
WARNING: Start of line didn't match any expected keyword.
WARNING: Problem on line 200: 'f.write("{}={}\\n".format(key, val))'
WARNING: Start of line didn't match any expected keyword.
WARNING: Problem on line 203: 'def parse_env_file(filepath: str) -> Dict[str, str]:'
WARNING: Start of line didn't match any expected keyword.
```

matt9j commented on init-odkx-sync-endpoint.py:
```diff
@@ -11,33 +11,37 @@
 
 app = typer.Typer(add_completion=False)
+localedir = os.path.join(os.path.abspath(os.path.dirname(__file__)), 'locale')
+translate = gettext.translation('handroll', localedir, fallback=True)
```
I think it would be better if the translation domain had some meaning to the project, maybe something like "odk-x-sync-endpoint-init" or something like that. It looks like "handroll" comes from an example.

matt9j commented on init-odkx-sync-endpoint.py:
```diff
@@ -57,73 +61,73 @@ def run_interactive_config():
-def run_cache_setup(envParam) -> bool:
+def run_cache_setup(envParam: Dict[str, str]) -> bool:
 
-    env:Dict[str, str] = envParam
```
Unrelated change in the commit. Okay for now, but in general it's better to try and keep each commit isolated to one specific set of changes.

matt9j commented on init-odkx-sync-endpoint.py:
```diff
@@ -57,73 +61,73 @@ def run_interactive_config():
+    enforce_https = typer.confirm(_("enforce https?"), default=True)
 
-    typer.echo("Would you like to enforce HTTPS? We recommend yes.")
-    enforce_https = is_enforce_https()
```
I think you want to keep is_enforce_https() a separate function to allow mocking it during testing, right?

matt9j commented on init-odkx-sync-endpoint.py:
```diff
@@ -57,73 +61,73 @@ def run_interactive_config():
     save_progress('enforce_https', enforce_https)
 
-    typer.echo(f"Enforcing https: {enforce_https}")
+    typer.echo(_("Enforcing https: {enforce_https}"))
```
Missing the call to .format() here

matt9j reviewed (changes requested):
Generally looks good! A couple small changes needed (see review) but the overall contribution structure is good!

matt9j commented:
Also a minor note, it's "internationalization", not "internalization"-- It's the same root as "international", like between nations. Internalization means the process of making something internal/inside, but we want to make the code international instead : )

jesse5555 committed:
add ability to support Internationalisation

This commit uses python's locale module to get the default language set in your system and the default gettext module to load the messages basing on the provided locale.  In case there is no default language set, English is taken as default and the user is notified about this.

solved french not working

solved mypy issue

remove pytest cache

jesse5555 commented:
> Also a minor note, it's "internationalization", not "internalization"-- It's the same root as "international", like between nations. Internalization means the process of making something internal/inside, but we want to make the code international to work in many different countries. I know it's a mouthful to pronounce haha : )

sorry for the typo. i actually hadnt have realised it until you spotted it out.  Thanks @matt9j

jesse5555 commented on tests/test_https.py:
```diff
@@ -48,3 +50,23 @@ def test_run_cache_setup(mocker):
+    mocker.patch('init-odkx-sync-endpoint.setup_certbot_certificate', return_value=None)
+    mocker.patch('init-odkx-sync-endpoint.setup_manual_certificate', return_value=None)
+
+    jw.setup_translator('fr')
```
@matt9j this must be changing the inputs to French.  Kindly share @matt9j if you know what i might be missing here

jesse5555 commented:
hey @matt9j @wbrunette 
Every thing works perfect now.  

I kindly request you to have a look.

matt9j reviewed (approved):
Looks great to me!
